### PR TITLE
SVG: optimize and refactor group creation code

### PR DESCRIPTION
**Pull request by timvandermeij**
This patch series brings down the number of transform groups for the Tracemonkey paper from 2790 to 1271, resulting in a much lighter DOM and faster rendering/scrolling. Refer to the commit messages for more information.

### CODE REVIEW

File: src/display/svg.js [hunk 1]
```diff
@@ -464,14 +462,9 @@ var SVGGraphics = (function SVGGraphicsClosure() {
...
         this.transformMatrix = IDENTITY_MATRIX;
-        this.pgrp = document.createElementNS(NS, 'svg:g'); // Parent group
-        this.pgrp.setAttributeNS(null, 'transform', pm(viewport.transform));
-        this.tgrp = document.createElementNS(NS, 'svg:g'); // Transform group
-        this.tgrp.setAttributeNS(null, 'transform', pm(this.transformMatrix));
         this.defs = document.createElementNS(NS, 'svg:defs');
-        this.pgrp.appendChild(this.defs);
-        this.pgrp.appendChild(this.tgrp);
-        this.svg.appendChild(this.pgrp);
+        this.svg.setAttributeNS(null, 'transform', pm(viewport.transform));
```
Comment (yurydelendik) [inline src/display/svg.js]: this does not work for chrome -- we will need to create a root group and use it as this.svg ?

File: src/display/svg.js [hunk 2]
```diff
@@ -1200,8 +1158,30 @@ var SVGGraphics = (function SVGGraphicsClosure() {
...
+      if (!this.tgrp) {
+        this.tgrp = document.createElementNS(NS, 'svg:g');
+        this.tgrp.setAttributeNS(null, 'transform', pm(this.transformMatrix));
+      }
+    },
+
+    /**
+     * @private
+     */
+    _finishGroup: function SVGGraphics_finishGroup() {
```
Comment (yurydelendik) [inline src/display/svg.js]: _finishGroup can be removed, see https://github.com/yurydelendik/pdf.js/commit/f4c8099678a0d19f1e48c891aa55c2581931495f

**Review by yurydelendik — CHANGES_REQUESTED** [review]
That is a really good start. I think we need to backout some changes in the viewer and avoid a regression in chrome.

**Comment by timvandermeij** [comment]
I have updated the commits based on the review comments.

File: src/display/svg.js (hunk 1 above)
Comment (timvandermeij) [inline src/display/svg.js]: Fixed in the new commits.

File: src/display/svg.js (hunk 2 above)
Comment (timvandermeij) [inline src/display/svg.js]: Fixed in the new commits. Nice find!

**Review by yurydelendik — APPROVED** [review]
Thank you for the patch.

**Pull request merged**